thirdeye-ui/src/app/components/anomalies-create/create-anomaly-properties-form/create-anomaly-properties-form.component.tsx

```diff
@@ -0,0 +1,249 @@
+                                    false,
+                                    undefined,
+                                    true,
+                                    "_blank"
+                                )}
+                        </>
+                    }
+                />
+
+                {/* Since these fields are read-only, only show them when the alert is loaded */}
+                {!!formFields.alert &&
+                    (
+                        Object.keys(
+                            readOnlyFormFields
+                        ) as (keyof CreateAnomalyReadOnlyFormFields)[]
+                    ).map(
+                        (
+                            readOnlyKey: keyof CreateAnomalyReadOnlyFormFields
+                        ) => (
+                            <InputSection
+                                inputComponent={
+                                    <TextField
+                                        fullWidth
+                                        required
+                                        InputProps={{
+                                            margin: "none",
+                                            readOnly: true,
+                                        }}
+                                        name={readOnlyKey}
+                                        title="These properties are derived from the selected alert"
+                                        type="string"
+                                        value={
+                                            readOnlyFormFields[readOnlyKey] ||
+                                            ""
+                                        }
+                                        variant="outlined"
+                                    />
+                                }
+                                key={readOnlyKey}
+                                label={formLabels[readOnlyKey]}
+                            />
+                        )
+                    )}
+
+                <LoadingErrorStateSwitch
+                    isError={enumerationItemsStatus === ActionStatus.Error}
+                    isLoading={enumerationItemsStatus === ActionStatus.Working}
+                    loadingState={
+                        <InputSection
+                            inputComponent={
+                                <SkeletonV1
+                                    height={50}
+                                    variant="rect"
+                                    width="100%"
+                                />
+                            }
+                            label={formLabels.enumerationItem}
+                        />
+                    }
+                >
+                    {!!(
+                        showEnumerationItemsField &&
+                        enumerationItemsForAlert &&
+                        enumerationItemsForAlert.length > 0
+                    ) && (
```

Comment:
Do we need `!!`?

Reply:
The reason for using `!!` is to coerce the left part of the `&&` into being a `boolean` type, otherwise it tends to sometimes render as a `0` if it is falsy

[Stackoverflow thread](https://stackoverflow.com/questions/53048037/react-showing-0-instead-of-nothing-with-short-circuit-conditional-component)

![image](https://user-images.githubusercontent.com/20799363/223927753-998c96a2-d49f-4b79-874d-6692a43a6734.png)